src/include/index/skiplist.h

```diff
@@ -44,24 +45,49 @@ class SkipList {
   ///////////////////////////////////////////////////////////////////
   // Core components
   ///////////////////////////////////////////////////////////////////
-  SkipListNode *skip_list_head_;
+  SkipListBaseNode *skip_list_head_;
   EpochManager epoch_manager_;
   NodeManager node_manager_;
   bool duplicate_support_;
   int GC_Interval_;
 
+  std::vector<SkipListInnerNode *> Search(const KeyType &key) {
+    return SearchFrom(key, skip_list_head_);
+  }
+
+  std::vector<SkipListInnerNode *> SearchFrom(const KeyType &key,
+                                              const SkipListBaseNode *Node) {
+    return std::vector<SkipListInnerNode *>{};
+  }
+
+  bool InsertNode(const KeyType &key, const ValueType &value) { return false; }
+
+  SkipListInnerNode *DeleteNode(const KeyType &key) { return nullptr; }
```

Comment:
If it returns a single node in a single list that is deleted, this node actually can only be deleted the next time someone reaches it. So returning the node doesn't seem necessary to me. 

I think delete requires us to send the EpochNode into the search key and let search process delete the inner node.

Reply:
Actually the node it return is already physically deleted from the skip-list. When it's physically deleted, we need this pointer for gc, right?